TypeError:  _blank() takes 1 positional

I am having this error:

```python
    def _run_request(self, *args, **kwargs):
>       return self._do_request(*args, **kwargs)
E       TypeError: _blank() takes 1 positional argument but 2 were given

/usr/local/lib/python3.10/site-packages/sanic_testing/testing.py:68: TypeError
```

Test function:
```

@pytest.mark.asyncio()
async def test_index(test_cli, app):
    resp = await test_cli.get('/')
    assert resp.status_code == 200
```

Versions:

```
pytest-sanic==1.9.1
sanic==21.6.2
sanic-babel==0.3.0
sanic-base-extension==0.1.1
sanic-devtools==0.1.0
sanic-jwt==1.7.0
sanic-routing==0.7.2
sanic-testing==0.7.0
```
This fixed it for me:
https://github.com/pawelkoston/sanic-testing/commit/9bb58429515c4006e4bf55bb865ae0d5c9e794ab

--- a/sanic_testing/testing.py
+++ b/sanic_testing/testing.py
@@ -48,7 +48,7 @@ class TestingResponse(httpx.Response):
         return self._json
 
 
-def _blank(self):
+def _blank(*_, **__):
     ...
 
 
@@ -352,6 +352,8 @@ class SanicASGITestClient(httpx.AsyncClient):
         self.sanic_app.router.reset()
         self.sanic_app.signal_router.reset()
         await self.sanic_app._startup()  # type: ignore
+        await self.sanic_app._server_event("init", "before")
+        await self.sanic_app._server_event("init", "after")
 
         if not url.startswith(
             ("http:", "https:", "ftp:", "ftps://", "//", "ws:", "wss:")
@@ -367,6 +369,9 @@ class SanicASGITestClient(httpx.AsyncClient):
 
         self.gather_request = gather_request
         response = await super().request(method, url, *args, **kwargs)
+
+        await self.sanic_app._server_event("shutdown", "before")
+        await self.sanic_app._server_event("shutdown", "after")
 
         response.__class__ = TestingResponse
 

--- a/tests/test_asgi_client.py
+++ b/tests/test_asgi_client.py
@@ -28,3 +28,35 @@ async def test_websocket_route(app):
 
     await app.asgi_client.websocket("/ws")
     assert ev.is_set()
+
+
+@pytest.mark.asyncio
+async def test_listeners(app):
+    listeners = []
+    available = (
+        "before_server_start",
+        "after_server_start",
+        "before_server_stop",
+        "after_server_stop",
+    )
+
+    @app.before_server_start
+    async def before_server_start(*_):
+        listeners.append("before_server_start")
+
+    @app.after_server_start
+    async def after_server_start(*_):
+        listeners.append("after_server_start")
+
+    @app.before_server_stop
+    async def before_server_stop(*_):
+        listeners.append("before_server_stop")
+
+    @app.after_server_stop
+    async def after_server_stop(*_):
+        listeners.append("after_server_stop")
+
+    await app.asgi_client.get("/")
+
+    assert len(listeners) == 4
+    assert all(x in listeners for x in available)

--- a/tests/test_test_client.py
+++ b/tests/test_test_client.py
@@ -28,3 +28,34 @@ def test_websocket_route(app):
     request, response = app.test_client.websocket("/ws")
     assert response.opened is True
     assert ev.is_set()
+
+
+def test_listeners(app):
+    listeners = []
+    available = (
+        "before_server_start",
+        "after_server_start",
+        "before_server_stop",
+        "after_server_stop",
+    )
+
+    @app.before_server_start
+    async def before_server_start(*_):
+        listeners.append("before_server_start")
+
+    @app.after_server_start
+    async def after_server_start(*_):
+        listeners.append("after_server_start")
+
+    @app.before_server_stop
+    async def before_server_stop(*_):
+        listeners.append("before_server_stop")
+
+    @app.after_server_stop
+    async def after_server_stop(*_):
+        listeners.append("after_server_stop")
+
+    app.test_client.get("/")
+
+    assert len(listeners) == 4
+    assert all(x in listeners for x in available)

